snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb

```diff
@@ -0,0 +1,129 @@
+module Snapshot
+  class SimulatorLauncherBase
+    attr_accessor :collected_errors
+
+    # The number of times we failed on launching the simulator... sigh
+    attr_accessor :current_number_of_retries_due_to_failing_simulator
+    attr_accessor :launcher_config
+
+    def initialize(launcher_configuration: nil)
+      @launcher_config = launcher_configuration
+    end
+
+    def collected_errors
+      @collected_errors || []
+    end
+
+    def current_number_of_retries_due_to_failing_simulator
+      @current_number_of_retries_due_to_failing_simulator || 0
+    end
+
+    def prepare_for_launch(language, locale, launch_arguments)
+      screenshots_path = TestCommandGenerator.derived_data_path
+      FileUtils.rm_rf(File.join(screenshots_path, "Logs"))
+      FileUtils.rm_rf(screenshots_path) if launcher_config.clean
+      FileUtils.mkdir_p(screenshots_path)
+
+      FileUtils.mkdir_p(CACHE_DIR)
+      FileUtils.mkdir_p(SCREENSHOTS_DIR)
+
+      File.write(File.join(CACHE_DIR, "language.txt"), language)
+      File.write(File.join(CACHE_DIR, "locale.txt"), locale || "")
+      File.write(File.join(CACHE_DIR, "snapshot-launch_arguments.txt"), launch_arguments.last)
+
+      prepare_simulators_for_launch(language: language, locale: locale)
+    end
+
+    def prepare_simulators_for_launch(language: nil, locale: nil)
+      # Kill and shutdown all currently running simulators so that the following settings
+      # changes will be picked up when they are started again.
+      Snapshot.kill_simulator # because of https://github.com/fastlane/snapshot/issues/337
+      `xcrun simctl shutdown booted &> /dev/null`
+
+      Fixes::SimulatorZoomFix.patch
```

Comment:
It might be worth evaluating if we still need those workarounds, probably not in the scope of this PR, but something to keep in mind.